nsxt/resource_nsxt_ns_service_group.go

```diff
@@ -0,0 +1,178 @@
+/* Copyright © 2018 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	api "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/common"
+	"github.com/vmware/go-vmware-nsxt/manager"
+	"log"
+	"net/http"
+)
+
+func resourceNsxtNsServiceGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtNsServiceGroupCreate,
+		Read:   resourceNsxtNsServiceGroupRead,
+		Update: resourceNsxtNsServiceGroupUpdate,
+		Delete: resourceNsxtNsServiceGroupDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"revision": getRevisionSchema(),
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "Description of this resource",
+				Optional:    true,
+			},
+			"display_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The display name of this resource. Defaults to ID if not set",
+				Optional:    true,
+				Computed:    true,
+			},
+			"tag": getTagsSchema(),
+			"members": &schema.Schema{
+				Type:        schema.TypeSet,
+				Description: "List of NSService or NSServiceGroup resources that can be added as members to an NSServiceGroup",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required: true,
+			},
+		},
+	}
+}
+
+func getResourceReferencesFromStringsSet(d *schema.ResourceData, schemaAttrName string) []common.ResourceReference {
+	targetIds := interface2StringList(d.Get(schemaAttrName).(*schema.Set).List())
+	var referenceList []common.ResourceReference
+	for _, targetID := range targetIds {
+		elem := common.ResourceReference{
+			IsValid:    true,
+			TargetId:   targetID,
+			TargetType: "NSService",
```

Comment:
when service group membership is supported, do we plan to run a get call for each member, in order to understand if its a service or a group? if not, we should solve it on user level.

Reply:
I think it should be solved in the user level, but I do not want to over complicate it now, since we do not know if it will ever be supported.
If it will be - we can add another parameter here for groups